packages/dds/tree/src/test/simple-tree/object.spec.ts

```diff
@@ -41,161 +45,199 @@ export function testObjectPrototype(proxy: object, prototype: object) {
+					// eslint-disable-next-line no-prototype-builtins -- compatibility test
+					assert.equal(asObject.isPrototypeOf(asObject), false);
 				});
+			});
 
-				describe("methods inherited from Object.prototype", () => {
-					it(`${pretty(initialTree)}.isPrototypeOf(Object.create(root)) -> true`, () => {
-						const root = hydrate(schema, initialTree);
-						const asObject = root as object;
-						// eslint-disable-next-line no-prototype-builtins -- compatibility test
-						assert.equal(asObject.isPrototypeOf(Object.create(asObject)), true);
-					});
+			describe(`${pretty(initialTree)}.propertyIsEnumerable`, () => {
+				for (const key of Object.getOwnPropertyNames(initialTree)) {
+					const expected = Object.prototype.propertyIsEnumerable.call(initialTree, key);
 
-					it(`${pretty(initialTree)}.isPrototypeOf(root) -> false`, () => {
+					it(`${key} -> ${expected}`, () => {
 						const root = hydrate(schema, initialTree);
 						const asObject = root as object;
 						// eslint-disable-next-line no-prototype-builtins -- compatibility test
-						assert.equal(asObject.isPrototypeOf(asObject), false);
+						assert.equal(asObject.propertyIsEnumerable(key), expected);
 					});
-				});
-
-				describe(`${pretty(initialTree)}.propertyIsEnumerable`, () => {
-					for (const key of Object.getOwnPropertyNames(initialTree)) {
-						const expected = Object.prototype.propertyIsEnumerable.call(
-							initialTree,
-							key,
-						);
-
-						it(`${key} -> ${expected}`, () => {
-							const root = hydrate(schema, initialTree);
-							const asObject = root as object;
-							// eslint-disable-next-line no-prototype-builtins -- compatibility test
-							assert.equal(asObject.propertyIsEnumerable(key), expected);
-						});
-					}
-				});
-			}
-		});
+				}
+			});
+		}
+	});
 
-		function test1(fn: (subject: object) => unknown) {
-			for (const { schema, initialTree } of testCases) {
-				const real = structuredClone(initialTree) as object;
-				const expected = fn(real);
+	function test1(fn: (subject: object) => unknown) {
+		for (const { schema, initialTree } of testCases) {
+			const real = structuredClone(initialTree) as object;
+			const expected = fn(real);
 
-				it(`${pretty(real)} -> ${pretty(expected)}`, () => {
-					const proxy = hydrate(schema, initialTree);
-					const actual = fn(proxy as object);
-					assert.deepEqual(actual, expected);
-				});
-			}
+			it(`${pretty(real)} -> ${pretty(expected)}`, () => {
+				const proxy = hydrate(schema, initialTree);
+				const actual = fn(proxy as object);
+				assert.deepEqual(actual, expected);
+			});
 		}
+	}
 
+	describe("keys", () => {
 		describe("Object.keys", () => {
 			test1((subject) => Object.keys(subject));
 		});
 
-		describe("Object.values", () => {
-			// eslint-disable-next-line @typescript-eslint/no-unsafe-return
-			test1((subject) => Object.values(subject));
+		describe("Reflect.ownKeys", () => {
+			test1((subject) => Reflect.ownKeys(subject));
```

Comment:
I can't tell what this is doing, but I think we want a unit tests for empty and full optional fields, value fields, and customized schema session local properties.